packages/telestion-client-core/src/hooks/abstractions/use-size.ts

```diff
@@ -0,0 +1,41 @@
+import { RefObject, useLayoutEffect, useRef, useState } from 'react';
+import useResizeObserver from '@react-hook/resize-observer';
+
+/**
+ * React Hook that determines the size of target {@link RefObject}
+ * and returns the target object and its size as a {@link DOMRect}.
+ *
+ * @example
+ * ```ts
+ * export function RootComponent() {
+ *     const [target, targetSize] = useSize<HTMLDivElement>();
+ *     const [size, setSize] = useState<DOMRect>();
+ *
+ *     useEffect(() => {
+ *         if (targetSize) setSize(targetSize);
+ *     }, [targetSize]);
+ *
+ *     return (
+ *         <div ref={target}>
+ *             {size}
+ *         </div>
+ *     );
+ * }
+ * ```
+ */
+export function useSize<T extends HTMLElement>(): readonly [
```

Comment:
Is there a necessity for this to be a part of the core? The core is typically meant for environment-agnostic things, meaning the `HTMLElement` reference makes this an oddity. If there's a reason for this to be in the core, that reason/exception should be documented in a comment. Thanks!